callBackPractice/callBackPractice.js

```diff
@@ -26,45 +30,86 @@ firstItem(foods, (firstItem) => {
 
 // Write a function called getLength that passes the length of the array into the callback
 
+const getLength = (arr, cb) => {
+	cb(arr.length);
+};
+
 getLength(foods, (length) => {
   console.log(`The length of the array is ${length}.`);
 });
 
 // Write a function called last which passes the last item of the array into the callback
 
+const last = (arr , cb) => {
+	cb(arr[(arr.length -1)]);
+};
+
 last(foods, (lastItem) => {
   console.log(`The last item in the array is ${lastItem}.`);
 });
 
 // Write a function called sumNums that adds two numbers and passes the result to the callback
 
+const sumNums = (num1, num2, cb) => {
+	cb(num1 + num2);
+};
 
 sumNums(5, 10, (sum) => {
   console.log(`The sum is ${sum}.`);
 });
 
 // Write a function called multiplyNums that adds two numbers and passes the result to the callback
 
+const multiplyNums = (num1, num2, cb) {
+	cb(num1 * num2);
+};
+
 multiplyNums(5, 10, (product) => {
   console.log(`The product is ${product}.`);
 });
 
 // Write a function called contains that checks if an item is present inside of the given array.
 // Pass true to the callback if it is, otherwise pass false
 
+const contains (array, str, cb) {
+	for (let i = 0; i < array.length; i++) {
+		if (array[i] === str) {
+			cb(true);
+		}
+	}
+	cb(false);
+};
+
 contains(foods, 'ribeye', (result) => {
   console.log(result ? 'ribeye is in the array' : 'ribeye is not in the array');
 });
 
 // Write a function called removeDuplicates that removes all duplicate values from the given array.
 // Pass the array to the callback function.  Do not mutate the original array.
 
+const removeDuplicates = (array, cb) => {
+	const changedArray = array.sort();
+	for (let i = 0; i < changedArray.length; i++) {
+		if (changedArray[i] === changedArray[i + 1]) {
+			changedArray.splice(i + 1, 1);
+		}
+	}
+	cb(changedArray);
+};
+
 removeDuplicates(foods, (uniqueFoods) => {
   console.log(`foods with duplicates removed: ${uniqueFoods}`);
 });
 
 // Write a function called forEach that iterates over the provided array and passes the value and index into the callback.
 
+const forEach = (array, cb) => {
+	//So close, works but doesn't. Meaning if you have duplicates it doesn't work like it should.
+	//Was having issues getting .entries() to work.
+	for (let i = 0; i < array.length; i++) {
+		cb(array[i], array.indexOf(array[i]));
```

Comment:
All you needed for this algorithm was to pass the elements of the array, and the index into the callback. I like where you're going for the remove duplicates but wasn't a necessary step! Nice job Matt!